pages/orders/[orderId]/labels.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  Button,
+  Checkbox,
+  CheckboxLabel,
+  Form,
+  FormGroup,
+  Grid,
+  GridItem,
+  Input,
+  Select,
+  Text,
+} from '@bigcommerce/big-design';
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+const StyledGrid = styled(Grid)`
+  grid-template: 'form print' auto / 1fr auto;
+
+  > .bd-button {
+    position: sticky;
+    top: 0;
+  }
+`;
+
+const OrderLabelPage: React.FC = () => {
+  const [shipping, setShipping] = useState('other');
+  const [shippingType, setShippingType] = useState<string>();
+  const [checked, setChecked] = useState(false);
+
+  const toggleChecked = () => {
+    setChecked(!checked);
+  };
+
+  useEffect(() => {
+    if (shipping === 'other') {
+      setShippingType(undefined);
+    }
+  }, [shipping]);
+
+  return (
+    <StyledGrid paddingBottom="xSmall" paddingHorizontal="xxSmall">
+      <GridItem gridArea="form" marginRight="medium">
+        <Form fullWidth={false}>
+          <FormGroup>
+            <Select
+              label="How would you like to ship?"
+              onOptionChange={setShipping}
+              options={[
+                { value: 'fedex', content: 'FedEx' },
+                { value: 'other', content: 'Other' },
+              ]}
+              required
+              value={shipping}
+            />
+          </FormGroup>
+          <FormGroup>
+            <Select
+              disabled={shipping === 'other'}
+              label="How would you like to ship?"
+              onOptionChange={setShippingType}
+              options={[
+                { value: 'flatrate', content: 'Flat rate' },
+                { value: 'variable', content: 'Variable' },
+              ]}
+              placeholder="Select or search a tracking carrier in the list"
+              required
+              value={shippingType}
+            />
+          </FormGroup>
+          <FormGroup>
+            <Input label="Tracking ID" required />
+          </FormGroup>
+          <FormGroup>
+            <Input defaultValue="Free Shipping" label="Shipping method description" required />
+          </FormGroup>
+          <FormGroup>
+            <Checkbox
+              checked={checked}
+              label={
+                <CheckboxLabel>
+                  Update order status to{' '}
+                  <Text as="span" bold>
+                    Shipped
+                  </Text>
+                  , and notify the customer via email.
+                </CheckboxLabel>
+              }
+              onClick={toggleChecked}
+            />
+          </FormGroup>
+        </Form>
+      </GridItem>
+      <Button>Print label</Button>
+    </StyledGrid>
+  );
+};
+
+export default OrderLabelPage;
```

Comment:
TBH, I dislike this page and may consider removing before merging. I primarily added this showcase a smaller modal.